src/doc/reference.md

```diff
@@ -3652,47 +3652,71 @@ sites are:
 
 * `let` statements where an explicit type is given.
 
-    In `let _: U = e;`, `e` is coerced to have type `U`.
+   For example, `128` is coerced to have type `i8` in the following:
+
+   ```rust
+   let _: i8 = 128;
+   ```
```

Comment:
I used 128 deliberately to show that the value *is* coerced into a type it can't fit in.

As for your other point, I also didn't of it as a coercion, but that's what the reference says.